nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/test/java/org/nuxeo/ecm/platform/comment/AbstractTestCommentNotification.java

```diff
@@ -140,26 +150,73 @@ public void shouldNotifyEventWhenRemoveComment() {
         // before subscribing, or previous event will be notified as well
         transactionalFeature.nextTransaction();
         // Reply
         captureAndVerifyCommentEventNotification(() -> {
             addSubscriptions("CommentAdded");
 
-            Comment reply = createComment(createdCommentDocModel);
+            Comment reply = createComment(commentedDocumentModel, ADMINISTRATOR, ANY_COMMENT_MESSAGE);
             DocumentModel replyDocumentModel = session.getDocument(new IdRef(reply.getId()));
             return session.getDocument(new IdRef(replyDocumentModel.getId()));
         }, COMMENT_ADDED, DOCUMENT_CREATED);
     }
 
-    private void addSubscriptions(String... notifications) {
+    protected void addSubscriptions(String... notifications) {
         NuxeoPrincipal principal = session.getPrincipal();
         String subscriber = NotificationConstants.USER_PREFIX + principal.getName();
         for (String notif : notifications) {
-            notificationService.addSubscription(subscriber, notif, commentedDocumentModel, false, principal, notif);
+            notificationManager.addSubscription(subscriber, notif, commentedDocumentModel, false, principal, notif);
+        }
+    }
+
+    @Test
+    @Deploy("org.nuxeo.ecm.platform.comment:OSGI-INF/notification-subscription-contrib.xml")
+    public void testAutoSubscribingOnlyOnceToNewComments() {
+        String john = "john";
+        String johnSubscription = NotificationConstants.USER_PREFIX + john;
+        createUser(john);
+        List<String> subscriptions = notificationManager.getSubscriptionsForUserOnDocument(johnSubscription,
+                commentedDocumentModel);
+        assertEquals(0, subscriptions.size());
+        createComment(commentedDocumentModel, john, "Test message");
+        transactionalFeature.nextTransaction();
+        commentedDocumentModel = session.getDocument(commentedDocumentModel.getRef());
+        subscriptions = notificationManager.getSubscriptionsForUserOnDocument(johnSubscription, commentedDocumentModel);
+        List<String> expectedSubscriptions = List.of(COMMENT_ADDED_NOTIFICATION, COMMENT_UPDATED_NOTIFICATION);
+        assertEquals(expectedSubscriptions.size(), subscriptions.size());
+        assertTrue(subscriptions.containsAll(expectedSubscriptions));
+        for (String subscription : subscriptions) {
+            notificationManager.removeSubscription(johnSubscription, subscription, commentedDocumentModel);
         }
+        createComment(commentedDocumentModel, john, "Test message again");
+        transactionalFeature.nextTransaction();
+        subscriptions = notificationManager.getSubscriptionsForUserOnDocument(johnSubscription, commentedDocumentModel);
+        assertEquals(0, subscriptions.size());
+    }
+
+    protected int checkNotifications(CapturingEventListener listener, Comment createdComment, int offset,
```

Comment:
i think this method is not used as now we test mainly the subscriptions behaviour, is it right